test/e2e/nodeoutofdisk.go

```diff
@@ -0,0 +1,246 @@
+// 4. Request the divided capacity for each pod.
+// 5. Observe that 2 of the pods schedule onto the node whose disk is not full, and the remaining
+//    pod stays pending and does not schedule onto the nodes whose disks are full nor the node
+//    with the other two pods, since there is not enough free CPU capacity there.
+// 6. Recover disk space from one of the nodes whose disk space was previously filled. Arbritrarily
+//    choose that node to be node with index 1.
+// 7. Observe that the pod in pending status schedules on that node.
+//
+// Reason: Initially when the disk space is filled on all but one node, the last pod fails to
+//         schedule because the node with available disk space will have no available CPU
+//         capacity. Although there will be CPU capacity available on other nodes, their disks
+//         will be full. When we recover disk space on the other node, both CPU capacity and disk
+//         space becomes available on that node and the pending pod schedules there.
+var _ = Describe("NodeOutOfDisk", func() {
+	var c *client.Client
+	var unfilledNodeName, recoveredNodeName string
+	framework := Framework{BaseName: "node-outofdisk"}
+
+	BeforeEach(func() {
+		framework.beforeEach()
+		c = framework.Client
+
+		nodelist, err := listNodes(c, labels.Everything(), fields.Everything())
+		expectNoError(err, "Error retrieving nodes")
+		Expect(len(nodelist.Items)).To(BeNumerically(">", 1))
+
+		unfilledNodeName = nodelist.Items[0].Name
+		for _, node := range nodelist.Items[1:] {
+			fillDiskSpace(c, &node)
+		}
+	})
+
+	AfterEach(func() {
+		defer framework.afterEach()
+
+		nodelist, err := listNodes(c, labels.Everything(), fields.Everything())
+		expectNoError(err, "Error retrieving nodes")
+		Expect(len(nodelist.Items)).ToNot(BeZero())
+		for _, node := range nodelist.Items {
+			if unfilledNodeName == node.Name || recoveredNodeName == node.Name {
+				continue
+			}
+			recoverDiskSpace(c, &node)
+		}
+	})
+
+	It("runs out of disk space", func() {
+		unfilledNode, err := c.Nodes().Get(unfilledNodeName)
+		expectNoError(err)
+
+		By(fmt.Sprintf("Get CPU capacity on node %s", unfilledNode.Name))
+
+		milliCPU := unfilledNode.Status.Capacity.Cpu().MilliValue()
+		// Per pod CPU should be just enough to fit only (numNodeOODPods - 1) pods on the
+		// given node. We compute this value by dividing the available CPU capacity on the given
+		// node by (numNodeOODPods - 1) and subtracting ϵ from it.
+		podCPU := (milliCPU / (numNodeOODPods - 1)) - (milliCPU / 10)
+
+		ns := framework.Namespace.Name
+		podClient := c.Pods(ns)
+
+		By("Creating pods")
+
+		podNames := make([]string, 0)
+
+		for i := 0; i < numNodeOODPods; i++ {
+			name := fmt.Sprintf("pod-node-outofdisk-%d", i)
+			createOutOfDiskPod(c, ns, name, podCPU)
+			podNames = append(podNames, name)
+		}
+
+		By("Waiting for all but one pods to be scheduled")
+		// Wait for all pods except the last one to be running
```

Comment:
Please add
TODO: This is a bit fragile because there's no guarantee the scheduler will try to schedule the pods in order (thus there's no guarantee the last pod is the one that will stay pending).

I wouldn't bother fixing it right now.